platform/iphone/export/export_plugin.cpp

```diff
@@ -205,9 +203,9 @@ void EditorExportPlatformIOS::_fix_config_file(const Ref<EditorExportPreset> &p_
 		} else if (lines[i].find("$bundle_identifier") != -1) {
 			strnew += lines[i].replace("$bundle_identifier", p_preset->get("application/bundle_identifier")) + "\n";
 		} else if (lines[i].find("$short_version") != -1) {
-			strnew += lines[i].replace("$short_version", p_preset->get("application/short_version")) + "\n";
+			strnew += lines[i].replace("$short_version", GLOBAL_GET("application/config/version")) + "\n";
 		} else if (lines[i].find("$version") != -1) {
-			strnew += lines[i].replace("$version", p_preset->get("application/version")) + "\n";
+			strnew += lines[i].replace("$version", GLOBAL_GET("application/config/version")) + "\n";
```

Comment:
I think only `$short_version` (`CFBundleShortVersionString`) should be replaced with `application/config/version`.     
`$version` (`CFBundleVersion`) which is internal application version would be better to stay as export setting. 

In some cases if `CFBundleShortVersionString` get's changed (which it should after every release) there might be no reason to change `CFBundleVersion` so it can stay the same. But it should be changed in case of AppStore rejection, so the version displayed in AppStore does not get increased.

Reply:
Seems like the ideal case here is to have `version` kept in the export setting, while having an empty value default to the `short_version` one.